WebGPU bikeshed error:  LINE ~6751: No 'dfn' refs found for 'serializable objects' that are marked for export.

The bulid-validate-publish flow is failing.
E.g. https://github.com/gpuweb/gpuweb/actions/runs/3943724824/jobs/6748888303

make: *** Waiting for unfinished jobs....
LINE ~6751: No 'dfn' refs found for 'serializable objects' that are marked for export.
[=serializable objects=]
 ✘  Did not generate, due to fatal errors
make[1]: *** [Makefile:10: index.pre.html] Error 2

I think the problem is in spec/index.bs:

$ grep -n "serializable ob" index.bs 
6751:{{GPUPipelineError}} objects are [=serializable objects=].

--- a/spec/index.bs
+++ b/spec/index.bs
@@ -78,6 +78,7 @@ spec: WGSL; urlPrefix: https://gpuweb.github.io/gpuweb/wgsl/#
         text: pipeline-overridable constant identifier string; url: pipeline-overridable-constant-identifier-string
         text: pipeline-overridable constant default value; url: pipeline-overridable-constant-default-value
         text: interface of a shader stage; url: interface-of-a-shader
+        text: shader execution end; url: shader-execution-end
         text: shader stage output; url: shader-stage-outputs
         text: shader stage input; url: shader-stage-inputs
         text: builtin; url: built-in-values
@@ -6748,7 +6749,7 @@ enum GPUPipelineErrorReason {
         </ul>
 </dl>
 
-{{GPUPipelineError}} objects are [=serializable objects=].
+{{GPUPipelineError}} objects are [[HTML#serializable-objects|serializable objects]].
 
 <div algorithm="GPUPipelineError serialization steps" data-timeline=content>
     Their [=serialization steps=], given |value| and |serialized|, are:
@@ -12880,7 +12881,7 @@ interface GPUCanvasContext {
 </div>
 
 When WebGPU canvas contents are read using other Web APIs, like
-{{CanvasDrawImage/drawImage()}}, `texImage2D()`, `texSubImage2D()`,
+[[HTML#drawing-images|drawImage()]], `texImage2D()`, `texSubImage2D()`,
 {{HTMLCanvasElement/toDataURL()}}, {{HTMLCanvasElement/toBlob()}}, and so on,
 they [$get a copy of the image contents of a context$].
 
@@ -13628,7 +13629,7 @@ These operations are encoded within {{GPUComputePassEncoder}} as:
 <p class="note editorial">Editorial: describe the computing algorithm
 
 The [=device=] may become [=lose the device|lost=] if
-[[WGSL#shader-execution-end|shader execution does not end]]
+[=shader execution end|shader execution does not end=]
 in a reasonable amount of time, as determined by the user agent.
 
 ## Rendering ## {#rendering-operations}
@@ -13850,7 +13851,7 @@ clip space positions for [[#primitive-clipping]], as well as other data for the
         will only be processed once.
 
         The [=device=] may become [=lose the device|lost=] if
-        [[WGSL#shader-execution-end|shader execution does not end]]
+        [=shader execution end|shader execution does not end=]
         in a reasonable amount of time, as determined by the user agent.
 </div>
 
@@ -14268,7 +14269,7 @@ This stage produces a <dfn dfn>Fragment</dfn> for each [=RasterizationPoint=]:
         1. Invoke the fragment shader entry point described by |desc|.
 
             The [=device=] may become [=lose the device|lost=] if
-            [[WGSL#shader-execution-end|shader execution does not end]]
+            [=shader execution end|shader execution does not end=]
             in a reasonable amount of time, as determined by the user agent.
 
         1. If the fragment issued `discard`, return `null`.

--- a/tools/install-dependencies.sh
+++ b/tools/install-dependencies.sh
@@ -5,7 +5,7 @@ code=1
 for opt in "$@"; do
     case "$opt" in
         bikeshed)
-            pip3 install bikeshed==3.7.0
+            pip3 install bikeshed==3.11.1
             bikeshed update
             code=0
             ;;

--- a/wgsl/index.bs
+++ b/wgsl/index.bs
@@ -3353,7 +3353,7 @@ If a reference or pointer access is out of bounds, an
 [=Load Rule|Loads=] from an invalid reference return one of:
     * when the [=originating variable=] is a [=uniform buffer=] or a [=storage buffer=],
         the value from any [=memory locations|memory location(s)=]
-        of the {{GPUBuffer}} bound to the originating variable
+        of the WebGPU {{GPUBuffer}} bound to the originating variable
     * when the originating variable is not a [=uniform buffer=] or [=storage buffer=],
         a value from any [=memory locations|memory location(s)=] in the originating variable
     * the [=zero value=] for store type of the reference
@@ -3363,7 +3363,7 @@ If a reference or pointer access is out of bounds, an
 [=statement/assignment|Stores=] to an invalid reference may do one of:
     * when the [=originating variable=] is a [=uniform buffer=] or a [=storage buffer=],
         store the value to any [=memory locations|memory location(s)=] of the
-        {{GPUBuffer}} bound to the [=originating variable=]
+        WebGPU {{GPUBuffer}} bound to the [=originating variable=]
     * when the [=originating variable=] is not a [=uniform buffer=] or [=storage buffer=],
         store the value to any [=memory locations|memory locations(s)=] in the
         originating variable

--- a/wgsl/index.bs
+++ b/wgsl/index.bs
@@ -31,7 +31,7 @@ Former Editor: dan sinclair, Google https://www.google.com, dsinclair@google.com
 Abstract: Shading language for WebGPU.
 Markup Shorthands: markdown yes
 Markup Shorthands: biblio yes
-Markup Shorthands: idl no
+Markup Shorthands: idl yes
 Assume Explicit For: yes
 </pre>
 
@@ -3867,7 +3867,7 @@ Note: The channel transfer function for 8snorm maps {-128,...,127} to the floati
 The texel formats listed in the
 <dfn lt="storage-texel-formats">Texel Formats for Storage Textures</dfn> table
 correspond to the [[WebGPU#plain-color-formats|WebGPU plain color formats]]
-which support the WebGPU {{GPUTextureUsage/STORAGE}} usage.
+which support the WebGPU {{GPUTextureUsage/STORAGE_BINDING}} usage.
 These texel formats are used to parameterize the storage texture types defined
 in [[#texture-storage]].
 
@@ -11452,8 +11452,8 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in value.
       <td>u32
       <td style="width:50%">Sample index for the current fragment.
          The value is least 0 and at most `sampleCount`-1, where
-         {{GPURenderPipelineDescriptor/sampleCount}}
-         is the number of MSAA samples specified for the GPU render pipeline.
+         `sampleCount` is the MSAA sample {{GPUMultisampleState/count}}
+         specified for the GPU render pipeline.
          <br>See [[WebGPU#render-pipeline]].
 
   <tr><td rowspan=2><dfn noexport dfn-for="built-in values">`sample_mask`</dfn>

--- a/wgsl/index.bs
+++ b/wgsl/index.bs
@@ -6289,7 +6289,7 @@ See [[#sync-builtin-functions]].
           * Otherwise, 0.
        * If |e1| is the most negative value in |T|, and |e2| is -1:
           * It is a [=shader-creation error=] if |e1| and |e2| are [=const-expressions=].
-          * It is a [=pipeline-creation error=] if |e1| and |e2| are [=override-expressions].
+          * It is a [=pipeline-creation error=] if |e1| and |e2| are [=override-expressions=].
           * Otherwise, 0.
        * Otherwise, |e1|&nbsp;-&nbsp;[=truncate=](|e1|&nbsp;&div;&nbsp;|e2|)&nbsp;&times;&nbsp;|e2|
            where the quotient is computed as a real value.

--- a/wgsl/index.bs
+++ b/wgsl/index.bs
@@ -280,7 +280,7 @@ spec: WebGPU; urlPrefix: https://gpuweb.github.io/gpuweb/#
         text: shader-output mask; url: shader-output-mask
         text: framebuffer; url: framebuffer
         for: supported limits
-            text: maxComputeWorkgroupStorageSize; url: #dom-supported-limits-maxcomputeworkgroupstoragesize
+            text: maxComputeWorkgroupStorageSize; url: dom-supported-limits-maxcomputeworkgroupstoragesize
     type: abstract-op
         text: validating GPUProgrammableStage; url: abstract-opdef-validating-gpuprogrammablestage
 </pre>

--- a/wgsl/index.bs
+++ b/wgsl/index.bs
@@ -269,6 +269,20 @@ spec: Unicode Character Database for Unicode Version 14.0.0; urlPrefix: https://
 spec: UnicodeVersion14; urlPrefix: https://www.unicode.org/versions/Unicode14.0.0/UnicodeStandard-14.0.pdf
     type: dfn
         text: code point; url:
+spec: WebGPU; urlPrefix: https://gpuweb.github.io/gpuweb/#
+    type: dfn
+        text: GPU command; url: gpu-command
+        text: RasterizationPoint; url: rasterizationpoint
+        text: binding member; url: binding-member
+        text: binding resource type; url: binding-resource-type
+        text: binding type; url: binding-type
+        text: front-facing; url: front-facing
+        text: shader-output mask; url: shader-output-mask
+        text: framebuffer; url: framebuffer
+        for: supported limits
+            text: maxComputeWorkgroupStorageSize; url: #dom-supported-limits-maxcomputeworkgroupstoragesize
+    type: abstract-op
+        text: validating GPUProgrammableStage; url: abstract-opdef-validating-gpuprogrammablestage
 </pre>
 
 # Introduction # {#intro}
@@ -288,7 +302,7 @@ that run on the GPU.
 
 ## Technical Overview ## {#technical-overview}
 
-WebGPU issues a unit of work to the GPU in the form of a [[WebGPU#gpu-command|GPU command]].
+WebGPU issues a unit of work to the GPU in the form of a [=GPU command=].
 WGSL is concerned with two kinds of GPU commands:
 * a <dfn noexport>draw command</dfn> executes a [=GPURenderPipeline|render pipeline=]
     in the context of [=shader stage input|inputs=], [=shader stage output|outputs=], and attached [=resources=].
@@ -308,7 +322,7 @@ A shader comprises:
 
 Note: A WGSL program does not require an [=entry point=]; however, such a
 program cannot be executed by the API because an entry point is required to
-create a [[WebGPU#GPUProgrammableStage]].
+create a {{GPUProgrammableStage}}.
 
 When executing a shader stage, the implementation:
 * Computes the values of constants declared at [=module scope|module-scope=].
@@ -466,14 +480,14 @@ The events are:
 
 1. <dfn noexport>Shader module creation</dfn>
     * This occurs when the
-        [[WebGPU#dom-gpudevice-createshadermodule|WebGPU createShaderModule]] method
+        WebGPU {{GPUDevice/createShaderModule()}} method
         is called.
         The source text for a WGSL program is provided at this time.
 2. <dfn noexport>Pipeline creation</dfn>
     * This occurs when the
-        [[WebGPU#dom-gpudevice-createcomputepipeline|WebGPU createComputePipeline]] method
+        WebGPU {{GPUDevice/createComputePipeline()}} method
         or the
-        [[WebGPU#dom-gpudevice-createrenderpipeline|WebGPU createRenderPipeline]] method
+        WebGPU {{GPUDevice/createRenderPipeline()}} method
         is invoked.
         These methods use one or more previously created shader modules, together with other
         configuration information.
@@ -3339,7 +3353,7 @@ If a reference or pointer access is out of bounds, an
 [=Load Rule|Loads=] from an invalid reference return one of:
     * when the [=originating variable=] is a [=uniform buffer=] or a [=storage buffer=],
         the value from any [=memory locations|memory location(s)=]
-        of the [[WebGPU#buffers|WebGPU buffer]] bound to the originating variable
+        of the {{GPUBuffer}} bound to the originating variable
     * when the originating variable is not a [=uniform buffer=] or [=storage buffer=],
         a value from any [=memory locations|memory location(s)=] in the originating variable
     * the [=zero value=] for store type of the reference
@@ -3349,7 +3363,7 @@ If a reference or pointer access is out of bounds, an
 [=statement/assignment|Stores=] to an invalid reference may do one of:
     * when the [=originating variable=] is a [=uniform buffer=] or a [=storage buffer=],
         store the value to any [=memory locations|memory location(s)=] of the
-        [[WebGPU#buffers|WebGPU buffer]] bound to the [=originating variable=]
+        {{GPUBuffer}} bound to the [=originating variable=]
     * when the [=originating variable=] is not a [=uniform buffer=] or [=storage buffer=],
         store the value to any [=memory locations|memory locations(s)=] in the
         originating variable
@@ -3701,7 +3715,7 @@ A <dfn noexport>texture</dfn> is a collection of texels supporting special opera
 In WGSL, those operations are invoked via texture builtin functions.
 See [[#texture-builtin-functions]] for a complete list.
 
-A WGSL texture corresponds to a [[WebGPU#gputexture|WebGPU GPUTexture]].
+A WGSL texture corresponds to a WebGPU {{GPUTexture}}.
 
 A texture is either arrayed, or non-arrayed:
 
@@ -3770,7 +3784,7 @@ The <dfn>texture types</dfn> are the set of types defined in:
 A [=sampler=] is an opaque handle that controls how [=texel|texels=] are accessed
 from a sampled texture.
 
-A WGSL sampler maps to a [[WebGPU#gpusampler|WebGPU GPUSampler]].
+A WGSL sampler maps to a WebGPU {{GPUSampler}}.
 
 Texel access is controlled via several properties of the sampler:
 
@@ -3778,18 +3792,18 @@ Texel access is controlled via several properties of the sampler:
 :: Controls how texture boundaries and out-of-bounds
     coordinates are resolved.
     The addressing mode for each texture dimension can be set independently.
-    See [[WebGPU#enumdef-gpuaddressmode|WebGPU GPUAddressMode]].
+    See WebGPU {{GPUAddressMode}}.
 : filter mode
 :: Controls which texels are accessed to produce the final result.
     Filtering can either use the nearest texel or interpolate between multiple
     texels.
     Multiple filter modes can be set independently.
-    See [[WebGPU#enumdef-gpufiltermode|WebGPU GPUFilterMode]].
+    See WebGPU {{GPUFilterMode}}.
 : LOD clamp
 :: Controls the min and max levels of details that are accessed.
 : comparison
 :: Controls the type of comparison done for [=type/sampler_comparison|comparison sampler=].
-    See [[WebGPU#enumdef-gpucomparefunction|WebGPU GPUCompareFunction]].
+    See WebGPU {{GPUCompareFunction}}.
 : max anisotropy
 :: Controls the maximum anisotropy value used by the sampler.
 
@@ -3816,7 +3830,7 @@ A <dfn noexport>texel format</dfn> is characterized by:
 : <dfn noexport>channel format</dfn>
 :: The number of bits in the channel, and how those bits are interpreted.
 
-Each texel format in WGSL corresponds to a [[WebGPU#enumdef-gputextureformat|WebGPU GPUTextureFormat]]
+Each texel format in WGSL corresponds to a WebGPU {{GPUTextureFormat}}
 with the same name.
 
 Only certain texel formats are used in WGSL source code.
@@ -3853,7 +3867,7 @@ Note: The channel transfer function for 8snorm maps {-128,...,127} to the floati
 The texel formats listed in the
 <dfn lt="storage-texel-formats">Texel Formats for Storage Textures</dfn> table
 correspond to the [[WebGPU#plain-color-formats|WebGPU plain color formats]]
-which support the [[WebGPU#dom-gputextureusage-storage|WebGPU STORAGE]] usage.
+which support the WebGPU {{GPUTextureUsage/STORAGE}} usage.
 These texel formats are used to parameterize the storage texture types defined
 in [[#texture-storage]].
 
@@ -3928,7 +3942,7 @@ but potentially with a different representation.
 It can be read using [[#textureload|textureLoad]] or [[#textureSampleBaseClampToEdge|textureSampleBaseClampToEdge]] built-in functions,
 which handle these different representations opaquely.
 
-See [[WebGPU#gpu-external-texture|WebGPU &sect; GPUExternalTexture]].
+See [[WebGPU#gpu-external-texture]].
 
 ### Storage Texture Types ### {#texture-storage}
 
@@ -8882,7 +8896,7 @@ To create an [=entry point=], declare a [=user-defined function=] with a [=shade
 
 When configuring a [=pipeline=] in the WebGPU API,
 the entry point's function name maps to the `entryPoint` attribute of the
-[[WebGPU#GPUProgrammableStage]] object.
+WebGPU {{GPUProgrammableStage}} object.
 
 The entry point's [=formal parameters=] denote the stage's [=shader stage inputs=].
 The entry point's [=return value=], if specified, denotes the stage's [=shader stage outputs=].
@@ -9229,7 +9243,7 @@ Each resource variable [=shader-creation error|must=] be declared with both [=at
 attributes.
 Together with the shader's stage, these identify the binding address
 of the resource on the shader's pipeline.
-See [[WebGPU#pipeline-layout|WebGPU &sect; GPUPipelineLayout]].
+See [[WebGPU#pipeline-layout]].
 
 Two different resource variables in a [=shader=] [=shader-creation error|must
 not=] have the same [=attribute/group=] and [=attribute/binding=] values, when
@@ -9241,60 +9255,59 @@ WebGPU requires that a shader's resource interface match the [[WebGPU#pipeline-l
 using the shader.
 
 It is a [=pipeline-creation error=] if a WGSL variable in a resource interface is bound to an incompatible WebGPU
-[[WebGPU#binding-resource-type|resource type]] or
-[[WebGPU#binding-type|binding type]],
+[=binding resource type=] or [=binding type=],
 where compatibility is defined by the following table.
 <table class='data'>
   <caption>WebGPU binding type compatibility</caption>
   <thead>
     <tr><th>WGSL resource
-        <th>WebGPU<br>[[WebGPU#binding-resource-type|Resource type]]
-        <th>WebGPU [[WebGPU#binding-member|Binding member]]
-        <th colspan=2>WebGPU [[WebGPU#binding-type|Binding type]]
+        <th>WebGPU<br>[=binding resource type|resource type=]
+        <th>WebGPU [=binding member=]
+        <th colspan=2>WebGPU [=binding type=]
   </thead>
   <tr><td>[=uniform buffer=]
-      <td rowspan=3>[[WebGPU#dictdef-gpubufferbinding|GPUBufferBinding]]
-      <td rowspan=3>[[WebGPU#dom-gpubindgrouplayoutentry-buffer|buffer]]
+      <td rowspan=3>{{GPUBufferBinding}}
+      <td rowspan=3>{{GPUBindGroupLayoutEntry/buffer}}
       <td rowspan=3>GPUBufferBindingType
-      <td>[[WebGPU#dom-gpubufferbindingtype-uniform|uniform]]
+      <td>{{GPUBufferBindingType/"uniform"}}
   <tr><td>[=storage buffer=] with [=access/read_write=] access
-      <td>[[WebGPU#dom-gpubufferbindingtype-storage|storage]]
+      <td>{{GPUBufferBindingType/"storage"}}
   <tr><td>[=storage buffer=] with [=access/read=] access
-      <td>[[WebGPU#dom-gpubufferbindingtype-read-only-storage|read-only-storage]]
+      <td>{{GPUBufferBindingType/"read-only-storage"}}
   <tr><td rowspan=2>sampler
-      <td rowspan=3>[[WebGPU#gpusampler|GPUSampler]]
-      <td rowspan=3>[[WebGPU#dom-gpubindgrouplayoutentry-sampler|sampler]]
+      <td rowspan=3>{{GPUSampler}}
+      <td rowspan=3>{{GPUBindGroupLayoutEntry/sampler}}
       <td rowspan=3>GPUSamplerBindingType
-      <td>[[WebGPU#dom-gpusamplerbindingtype-filtering|filtering]]
+      <td>{{GPUSamplerBindingType/"filtering"}}
   <tr>
-      <td>[[WebGPU#dom-gpusamplerbindingtype-non-filtering|non-filtering]]
+      <td>{{GPUSamplerBindingType/"non-filtering"}}
   <tr><td>sampler_comparison
-      <td>[[WebGPU#dom-gpusamplerbindingtype-comparison|comparison]]
+      <td>{{GPUSamplerBindingType/"comparison"}}
   <tr><td rowspan=5>sampled texture
-      <td rowspan=5>[[WebGPU#gputextureview|GPUTextureView]]
-      <td rowspan=5>[[WebGPU#dom-gpubindgrouplayoutentry-texture|texture]]
+      <td rowspan=5>{{GPUTextureView}}
+      <td rowspan=5>{{GPUBindGroupLayoutEntry/texture}}
       <td rowspan=5>GPUTextureSampleType
-      <td>[[WebGPU#dom-gputexturesampletype-float|float]]
+      <td>{{GPUTextureSampleType/"float"}}
   <tr>
-      <td>[[WebGPU#dom-gputexturesampletype-unfilterable-float|unfilterable-float]]
+      <td>{{GPUTextureSampleType/"unfilterable-float"}}
   <tr>
-      <td>[[WebGPU#dom-gputexturesampletype-sint|sint]]
+      <td>{{GPUTextureSampleType/"sint"}}
   <tr>
-      <td>[[WebGPU#dom-gputexturesampletype-uint|uint]]
+      <td>{{GPUTextureSampleType/"uint"}}
   <tr>
-      <td>[[WebGPU#dom-gputexturesampletype-depth|depth]]
+      <td>{{GPUTextureSampleType/"depth"}}
   <tr><td>[=write-only storage texture=]
-      <td>[[WebGPU#gputextureview|GPUTextureView]]
-      <td>[[WebGPU#dom-gpubindgrouplayoutentry-storagetexture|storageTexture]]
-      <td>GPUStorageTextureAccess
-      <td>[[WebGPU#dom-gpustoragetextureaccess-write-only|write-only]]
+      <td>{{GPUTextureView}}
+      <td>{{GPUBindGroupLayoutEntry/storageTexture}}
+      <td>{{GPUStorageTextureAccess}}
+      <td>{{GPUStorageTextureAccess/"write-only"}}
   <tr><td>[[#external-texture-type|external sampled texture]]
-      <td>[[WebGPU#gpuexternaltexture|GPUExternalTexture]]
-      <td>[[WebGPU#dom-gpubindgrouplayoutentry-externaltexture|externalTexture]]
+      <td>{{GPUExternalTexture}}
+      <td>{{GPUBindGroupLayoutEntry/externalTexture}}
       <td colspan=2>(not applicable)
 </table>
 
-See the [[WebGPU#abstract-opdef-validating-gpuprogrammablestage|WebGPU API]]
+See the [$validating GPUProgrammableStage|WebGPU API$]
 specification for interface validation requirements.
 
 # Language Extensions # {#language-extensions}
@@ -9438,7 +9451,7 @@ shader that goes beyond the specified limits.
             [=address spaces/workgroup=] address space
 
             For the purposes of this limit, [=bool=] has a size of 1 byte.
-        <td>[[WebGPU#dom-supported-limits-maxcomputeworkgroupstoragesize|16384]]
+        <td>[=supported limits/maxComputeWorkgroupStorageSize|16384=]
     <tr><td>Maximum number of elements in [=const-expression=] of [=array=] type<td>65535
 </table>
 
@@ -10572,7 +10585,7 @@ Quads can collaborate in some collective operations (see [[#derivatives]]).
 
 Ordinarily, [[WebGPU#fragment-processing|fragment processing]] creates one
 invocation of a fragment shader for each
-[[WebGPU#rasterizationpoint|RasterizationPoint]] produced by
+[=RasterizationPoint=] produced by
 [[WebGPU#rasterization|rasterization]].
 Sometimes there may be insufficient RasterizationPoints to fully populate a
 quad, for example at the edge of a graphics primitive.
@@ -11373,30 +11386,29 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in value.
       After homogeneous normalization (where each of the *x*, *y*, and *z* components
       are divided by the *w* component), the position is in the WebGPU normalized device
       coordinate space.
-      See [[WebGPU#coordinate-systems|WebGPU &sect; Coordinate Systems]].
+      See [[WebGPU#coordinate-systems]].
 
   <tr>
       <td>fragment
       <td>input
       <td>vec4&lt;f32&gt;
-      <td style="width:50%">Framebuffer position of the current fragment in [[WebGPU#rasterization|framebuffer space]].
+      <td style="width:50%">Framebuffer position of the current fragment in [=framebuffer=] space.
       (The *x*, *y*, and *z* components have already been scaled such that *w* is now 1.)
-      See [[WebGPU#coordinate-systems|WebGPU &sect; Coordinate Systems]].
+      See [[WebGPU#coordinate-systems]].
 
   <tr><td><dfn noexport dfn-for="built-in values">`front_facing`</dfn>
       <td>fragment
       <td>input
       <td>bool
-      <td style="width:50%">True when the current fragment is on a front-facing primitive.
+      <td style="width:50%">True when the current fragment is on a [=front-facing=] primitive.
          False otherwise.
-         See [[WebGPU#front-facing|WebGPU &sect; Front-facing]].
 
   <tr><td><dfn noexport dfn-for="built-in values">`frag_depth`</dfn>
       <td>fragment
       <td>output
       <td>f32
       <td style="width:50%">Updated depth of the fragment, in the viewport depth range.
-      See [[WebGPU#coordinate-systems|WebGPU &sect; Coordinate Systems]].
+      See [[WebGPU#coordinate-systems]].
 
   <tr><td><dfn noexport dfn-for="built-in values">`local_invocation_id`</dfn>
       <td>compute
@@ -11432,7 +11444,7 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in value.
       <td>vec3&lt;u32&gt;
       <td style="width:50%">The [=dispatch size=], `vec<u32>(group_count_x,
       group_count_y, group_count_z)`, of the compute shader
-      [[WebGPU#dom-gpucomputepassencoder-dispatch|dispatched]] by the API.
+      [[WebGPU#compute-pass-encoder-dispatch|dispatched]] by the API.
 
   <tr><td><dfn noexport dfn-for="built-in values">`sample_index`</dfn>
       <td>fragment
@@ -11440,9 +11452,9 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in value.
       <td>u32
       <td style="width:50%">Sample index for the current fragment.
          The value is least 0 and at most `sampleCount`-1, where
-         [[WebGPU#dom-gpurenderpipelinedescriptor-samplecount|sampleCount]]
+         {{GPURenderPipelineDescriptor/sampleCount}}
          is the number of MSAA samples specified for the GPU render pipeline.
-         <br>See [[WebGPU#render-pipeline|WebGPU &sect; GPURenderPipeline]].
+         <br>See [[WebGPU#render-pipeline]].
 
   <tr><td rowspan=2><dfn noexport dfn-for="built-in values">`sample_mask`</dfn>
       <td>fragment
@@ -11451,7 +11463,7 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in value.
       <td style="width:50%">Sample coverage mask for the current fragment.
          It contains a bitmask indicating which samples in this fragment are covered
          by the primitive being rendered.
-         <br>See [[WebGPU#sample-masking|WebGPU &sect; Sample Masking]].
+         <br>See [[WebGPU#sample-masking]].
 
   <tr>
       <td>fragment
@@ -11459,10 +11471,10 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in value.
       <td>u32
       <td style="width:50%">Sample coverage mask control for the current fragment.
          The last value written to this variable becomes the
-         [[WebGPU#shader-output-mask|shader-output mask]].
+         [=shader-output mask=].
          Zero bits in the written value will cause corresponding samples in
          the color attachments to be discarded.
-         <br>See [[WebGPU#sample-masking|WebGPU &sect; Sample Masking]].
+         <br>See [[WebGPU#sample-masking]].
 </table>
 
 <div class='example wgsl global-scope' heading="Declaring built-in values">
@@ -14802,8 +14814,8 @@ fn textureSampleBaseClampToEdge(t: T,
   >  *half_texel* = vec2(0.5) / vec2&lt;f32&gt;(textureDimensions(t))
 
   Note: The half-texel adjustment ensures that,
-  independent of the sampler's [[WebGPU#enumdef-gpuaddressmode|addressing]]
-  and [[WebGPU#enumdef-gpufiltermode|filter]] modes,
+  independent of the sampler's {{GPUAddressMode|addressing}}
+  and {{GPUFilterMode|filter}} modes,
   wrapping will not occur.
   That is, when sampling near an edge, the sampled texels
   will be at or adjacent to that edge, and not selected from the opposite edge.

